components/locid_transform/src/canonicalizer.rs

```diff
@@ -224,25 +218,25 @@ impl LocaleCanonicalizer {
     #[doc = icu_provider::gen_any_buffer_unstable_docs!(ANY, Self::new)]
     pub fn try_new_with_any_provider(
         provider: &(impl AnyProvider + ?Sized),
-    ) -> Result<LocaleCanonicalizer, LocaleTransformError> {
+    ) -> Result<Self, LocaleTransformError> {
         let expander = LocaleExpander::try_new_with_any_provider(provider)?;
-        Self::try_new_with_expander_unstable(&provider.as_downcasting(), expander)
+        Self::try_new_with_expander_with_any_provider(&provider, expander)
```

Comment:
Thought: This is correct but I guess it's the equivalent of `try_new_with_expander_compat(&provider.as_downcasting())`